src/CosmosDB/CosmosDB/Service/NewAzCosmosDBService.cs

```diff
@@ -60,9 +60,11 @@ public override void ExecuteCmdlet()
                 ResourceGroupName = resourceIdentifier.ResourceGroupName;
             }
 
-            ServiceResourceCreateUpdateParameters parameters = new ServiceResourceCreateUpdateParameters(
+            ServiceResourceCreateUpdateProperties properties = new ServiceResourceCreateUpdateProperties(
```

Comment:
In the latest iteration ServiceResourceCreateUpdateProperties is made an abstract class, which essentially means we cannot instantiate it manually from the code. All code references to it will be manually replaced by the child classes. The child classes are determined from the serviceType. In short, here we should remove the reference to ServiceResourceCreateUpdateProperties with ServiceType.SqlDedicatedGateway, and instead just instantiate the child class SqlDedicatedGatewayServiceResourceCreateUpdateProperties